lib/Conversion/ImportVerilog/ImportVerilog.cpp

```diff
@@ -260,14 +257,44 @@ LogicalResult ImportContext::preprocessVerilog(llvm::raw_ostream &os) {
         break;
     }
 
-    for (auto &diag : diagnostics) {
+    for (auto &diag : preprocessor.getDiagnostics()) {
       if (diag.isError()) {
         driver.diagEngine.issue(diag);
         return failure();
       }
     }
     os << output.str();
+    return success();
+  };
+
+  // Depending on whether the single-unit option is set, either add all source
+  // files to a single preprocessor such that they share define macros and
+  // directives, or create a separate preprocessor for each, such that each
+  // source file is in its own compilation unit.
+  auto optionBag = driver.createOptionBag();
+  if (driver.options.singleUnit == true) {
+    slang::BumpAllocator alloc;
+    slang::Diagnostics diagnostics;
+    slang::parsing::Preprocessor preprocessor(driver.sourceManager, alloc,
+                                              diagnostics, optionBag);
+    // Sources have to be pushed in reverse, as they form a stack in the
+    // preprocessor. Last pushed source is processed first.
+    for (auto &buffer : slang::make_reverse_range(driver.buffers))
```

Comment:
I'm curious about this via `llvm::reverse(...)` but LGTM.

Reply:
I tried that first, but ran into an ambiguous reference to `make_range` inside the reverse function (one provided by slang, one by llvm).